cmake/CompileFlags.cmake

```diff
@@ -68,14 +68,29 @@ macro(compile_flags)
             -fno-strict-aliasing
         )
 
-        add_compile_options(
-            # When we are a stable release (Release build + USE_ASSERTS not set),
-            # assertations are off, which trigger a lot of warnings. We disable
-            # these warnings for these releases.
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
-            "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
-        )
+        # When we are a stable release (Release build + USE_ASSERTS not set),
+        # assertations are off, which trigger a lot of warnings. We disable
+        # these warnings for these releases.
+        if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-parameter>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-but-set-variable>"
+            )
+        else (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+            add_compile_options(
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-variable>"
+                "$<${IS_STABLE_RELEASE}:-Wno-unused-parameter>"
+            )
+        endif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+
+        if (${OPTION_FORCE_COLORED_OUTPUT})
```

Comment:
`if (OPTION_FORCE_COLORED_OUTPUT)` should work here

Reply:
That said, why add it at all? Why is it "useful for Ninja" ? Big ol comment somewhere please

Reply:
> That said, why add it at all? Why is it "useful for Ninja" ? Big ol comment somewhere please

Ninja processes the output so the output from the compiler isn't directly to a terminal; hence, the default is non-coloured output.

Reply:
I figured that would be the reason, but I'd like to see a comment in the code explaining it

Reply:
Fixed!